sycl/source/detail/queue_impl.hpp

```diff
@@ -98,6 +103,31 @@ class queue_impl {
         MIsProfilingEnabled(has_property<property::queue::enable_profiling>()),
         MHasDiscardEventsSupport(MDiscardEvents &&
                                  (MHostQueue ? true : MIsInorder)) {
+    // The following commented section provides a guideline on how to use the
+    // TLS enabled mechanism to create a tracepoint and notify using XPTI. This
+    // is the prolog section and the epilog section will initiate the
+    // notification.
+    // #if XPTI_ENABLE_INSTRUMENTATION
+    //     /// This section of code is relying on scoped objects, so they cannot
+    //     be
+    //     /// encapsulated in a function
+    //     detail::tls_code_loc_t Tls;
+    //     auto TData = Tls.query();
+    //     xpti::framework::tracepoint_t TP(TData.fileName(),
+    //     TData.functionName(),
+    //                                      TData.lineNumber(),
+    //                                      TData.columnNumber    (), (void
+    //                                      *)this);
+    //     if (xptiTraceEnabled()) {
+    //       TP.stream(SYCL_STREAM_NAME)
+    //           .trace_type(xpti::trace_point_type_t::queue_create);
```

Comment:
IMO, this is way too verbose. I'd rather see something like

```
void xpti_notify(auto Callback) {
  #if XPTI_ENABLE_INSTRUMENTATION
  detail::tls_code_loc_t Tls;
  .... // rest of code here.
  Callback(getTEventFromTLS(TLS)); // To add callsite specific data
  ... // whatever common is needed after specific data
  #endif
}

void foo() {
  xpti_notify([](auto TEvent) { 
     xpti::addMetadata(TEvent, "src_memory_ptr", reinterpret_cast<size_t>(Src));
  });
}
```

Reply:
Yes, thank you for the suggestion. I have updated the code to reflect this.

Reply:
Thanks for the suggestion - have used this outline from you and some suggestions from Vlad to refactor this to be more concise.